theano/gof/destroyhandler.py

```diff
@@ -793,14 +794,14 @@ def fast_destroy(self, app):
         dm = getattr(app.op, 'destroy_map', None)
         if not dm:
             return
-        # inputs = sum(dm.values())  # list of app's destroyed inputs
-        inputs = dm.values()[0]
+        inputs = list(set(itertools.
+                          chain.from_iterable(dm.values())))   # list of app's destroyed inputs
```

Comment:
No the problem is that each item inside```dm.values()``` is itself a list. So for example : 
```dm = {0: [1, 2], 1:[0, 1]}```. In this case the inputs that are destroyed are 
```list(set(1, 2, 0, 1)) = [0, 1, 2]```.